### PR TITLE
feat: NGSCheckMate

### PR DESCRIPTION
**Pull request by tdayris**
This PR adds [NGSCheckMate](https://github.com/parklab/NGSCheckMate) to the list of available wrappers.

### QC

* [X] I confirm that I have followed the [documentation for contributing to `snakemake-wrappers`](https://snakemake-wrappers.readthedocs.io/en/stable/contributing.html).

While the contributions guidelines are more extensive, please particularly ensure that:
* [X] `test.py` was updated to call any added or updated example rules in a `Snakefile`
* [X] `input:` and `output:` file paths in the rules can be chosen arbitrarily
* [X] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`)
* [X] temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to 
* [X] the `meta.yaml` contains a link to the documentation of the respective tool or command under `url:`
* [X] conda environments use a minimal amount of channels and packages, in recommended ordering

## Release Notes

- **New Features**
  - Introduced new environment files (`environment.linux-64.pin.txt`, `environment.yaml`, `meta.yaml`) for easier setup and configuration of the NGS-Checkmate tool.
  - Added a wrapper script to streamline the processing of genomic data files.

- **Testing Enhancements**
  - Added a new test function to validate the functionality of the NGS-Checkmate tool with various input file types.

- **New Test Files**
  - Included multiple test files (`a.small.vcf`, `b.small.vcf`, `small.bed`, `small_genome.fasta`, `small_bwt.4.ebwt`, `small_genome.fasta.fai`) to support testing and validation processes.

### CODE REVIEW

File: bio/ngscheckmate/ncm/wrapper.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            raise ValueError("Mixed BAM/VCF/Fastq. Please provide only one file type.")
+
+    list_file = snakemake.input.get("list_file", list_file)
+    extra += f" --list {list_file} "
+
+    cmd = (
+        f"export NCM_REF='{snakemake.input.fasta}' && "
+        f"{launcher} {extra} --outdir {tempdir} "
+        f"-N snake_out {log} "
+    )
```
[the remark this answers is not in the document]
Comment (tdayris) [reply bio/ngscheckmate/ncm/wrapper.py:107]: If use wants to provide sub-commands through params, they are free to do it.

**Comment by schrins** [comment]
The wrapper looks good so far. There are a couple of typos, though, that should be corrected before merging. The exact positions are listed in the coderabbitai post from 3 days ago: https://github.com/snakemake/snakemake-wrappers/pull/3498#pullrequestreview-2458071101
1. Some function names contain `ngschekmate` instead of `ngscheckmate`
2. Some test output files have a misleading name, e.g. `bam_rscript.R` for a test about FASTQ files
3. The unused import `isdir` can be removed
4. Other bot suggestions like enhancing VCF files can be ignored because the tests are supposed to test the wrapper, not the underlying tool.

File: bio/ngscheckmate/ncm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  matrix: Path to samples matrix (metrics)
+  rscript: Path to the rscript used to produce the PDF plot
+  list_file: Path to the temporary list_file build for internal NGS-Checkmate processing. 
+params:
+  extra: Optional parameters besides IO and threading.
+notes: |
+  See a description of the list-file in the official documentation.
+  Pattern-file can be build with Snakemake-wrapper `ngscheckmate/makesnvpattern`.
+
+  Rememver, the pattern file is required for `fastq` input files. The BED intervals are required for BAM/VCF input files.
```
Comment (schrins) [inline bio/ngscheckmate/ncm/meta.yaml:23]: Remember, the pattern file is required for `fastq` input files. The BED intervals are required for BAM/VCF input files.
```suggestion
  Remember, the pattern file is required for `fastq` input files. The BED intervals are required for BAM/VCF input files.
```

File: test_wrappers.py [hunk 3]
```diff
@@ -4196,6 +4196,29 @@ def test_star_align(run):
...
+    )
+    run(
+        "bio/ngscheckmate/ncm",
+        ["snakemake", "--cores", "1", "fastq_paired_matched.txt", "--use-conda", "-F"],
+    )
+    
+    
+    
+
+
```
Comment (schrins) [inline test_wrappers.py:4220]:
```suggestion

```

**Review by schrins — APPROVED** [review]
Looks fine now.

File: bio/ngscheckmate/ncm/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  samples: Either a path to a list-file, or list of paths to alignment (BAM/SAM) files, or list of paths to variants files (VCF/BCF), or list of paths to FastQ files.
+  bed: Path to a interval (BED) file, required for VCF/BAM input.
+  pt: Path to a pattern file, required for Fastq input
+output:
+  pdf: Path to plot file (PDF formatted)
+  matched: Path to matched samples table
+  txt: Path to samples description
+  matrix: Path to samples matrix (metrics)
+  rscript: Path to the rscript used to produce the PDF plot
+  list_file: Path to the temporary list_file build for internal NGS-Checkmate processing.
```
Comment (fgvieira) [inline bio/ngscheckmate/ncm/meta.yaml:16]: This seems to be just a temporary file from the wrapper. If so, it should not be an output.

File: bio/ngscheckmate/ncm/wrapper.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    shell("echo '{cmd}' {log}")
+    shell(cmd)
+
+    shell("tree -srhf {tempdir} {log} ")
```
Comment (fgvieira) [inline bio/ngscheckmate/ncm/wrapper.py:111]: Debug print?
```suggestion
```

File: bio/ngscheckmate/ncm/wrapper.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    list_file = snakemake.input.get("list_file", list_file)
+    extra += f" --list {list_file} "
+
+    cmd = (
+        f"export NCM_REF='{snakemake.input.fasta}' && "
+        f"{launcher} {extra} --outdir {tempdir} "
+        f"-N snake_out {log} "
+    )
+    shell("echo '{cmd}' {log}")
+    shell(cmd)
```
Comment (fgvieira) [inline bio/ngscheckmate/ncm/wrapper.py:109]:
```suggestion
    shell("export NCM_REF={snakemake.input.fasta:q} && {launcher} {extra} --outdir {tempdir} -N snake_out {log}")
```

File: bio/ngscheckmate/ncm/wrapper.py [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+                is_paired=snakemake.params.get("paired", True),
+            )
+
+            # Only ncm_fastq.py handles multithreading
+            extra += f" --maxthread {snakemake.threads} "
+        else:
+            raise ValueError("Mixed BAM/VCF/Fastq. Please provide only one file type.")
+
+    list_file = snakemake.input.get("list_file", list_file)
+    extra += f" --list {list_file} "
```
Comment (fgvieira) [inline bio/ngscheckmate/ncm/wrapper.py:101]: Can this list also be an input? If so, can you add it to the `meta.yaml` file? And how does it clash with `input.samples`?

File: bio/ngscheckmate/ncm/wrapper.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+                output=list_file,
+                is_paired=snakemake.params.get("paired", True),
+            )
+
+            # Only ncm_fastq.py handles multithreading
+            extra += f" --maxthread {snakemake.threads} "
+        else:
+            raise ValueError("Mixed BAM/VCF/Fastq. Please provide only one file type.")
+
+    list_file = snakemake.input.get("list_file", list_file)
```
Comment (fgvieira) [inline bio/ngscheckmate/ncm/wrapper.py:100]: Can this list also be an input? If so, can you add it to the `meta.yaml` file? And how does it clash with `input.samples`?

File: bio/ngscheckmate/ncm/wrapper.py [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+# Pattern file is expected for FastQ files only
+pattern = snakemake.input.get("pattern", "")
+bed = snakemake.input.get("bed", "")
+if bed:
+    extra += f" --bedfile {bed} "
+
+with TemporaryDirectory() as tempdir:
+    samples = snakemake.input.get("samples")
+    list_file = f"{tempdir}/list_file.txt"
```
[the remark this answers is not in the document]
Comment (johanneskoester) [reply bio/ngscheckmate/ncm/wrapper.py:68]: This is not needed, as the files are already required by the wrapper.

**Pull request merged**